sound/soc/sof/topology.c

```diff
@@ -1900,6 +1921,18 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 		return ret;
 	}
 
+	/* bind widgets with their event handlers */
+	if (tw->event_type) {
+		ret = snd_soc_tplg_widget_bind_event(swidget->widget,
+						     sof_tplg_widget_ops,
+						     ARRAY_SIZE(sof_tplg_widget_ops),
+						     tw->event_type);
+		if (ret) {
+			dev_err(sdev->dev, "error: widget event binding failed\n");
+			return -EINVAL;
```

Comment:
as described above, maybe we can specialize this widget ops table and use for the detector, instead of having a generic table that only works for the Detector?

Reply:
sounds good. I will change this